crates/graphics/src/primitives/shaders/grid.wgsl

```diff
@@ -0,0 +1,93 @@
+#import luminol::translation as Trans  // 🏳️‍⚧️
+
+struct VertexOutput {
+    @builtin(position) clip_position: vec4<f32>,
+    @location(0) position: vec2<f32>,
+    // The fragment shader sees this as the position of the provoking vertex,
+    // which is set to the vertex at the right angle of every triangle
+    @location(1) @interpolate(flat) vertex_position: vec2<f32>,
+}
+
+struct Display {
+    pixels_per_point: f32,
+    inner_thickness_in_points: f32,
+    map_size: vec2<u32>,
+}
+
+// OpenGL and WebGL use the last vertex in each triangle as the provoking vertex, and
+// Direct3D, Metal, Vulkan and WebGPU use the first vertex in each triangle
+#ifdef LUMINOL_BACKEND_GL
+const QUAD_VERTICES: array<vec2f, 6> = array<vec2f, 6>(
+    vec2f(1., 0.),
+    vec2f(0., 1.),
+    vec2f(0., 0.), // Provoking vertex
+    
+    vec2f(0., 1.),
+    vec2f(1., 0.),
+    vec2f(1., 1.), // Provoking vertex
+);
+#else
+const QUAD_VERTICES: array<vec2<f32>, 6> = array<vec2<f32>, 6>(
+    vec2<f32>(0., 0.),
+    vec2<f32>(1., 0.),
+    vec2<f32>(0., 1.), // Provoking vertex
+    
+    vec2<f32>(1., 1.),
+    vec2<f32>(0., 1.),
+    vec2<f32>(1., 0.), // Provoking vertex
+);
+#endif
```

Comment:
The comments in the grid shader for which vertex is the provoking vertex are probably inaccurate. It should be like this:
```suggestion
// OpenGL and WebGL use the last vertex in each triangle as the provoking vertex, and
// Direct3D, Metal, Vulkan and WebGPU use the first vertex in each triangle
#ifdef LUMINOL_BACKEND_GL
const QUAD_VERTICES: array<vec2f, 6> = array<vec2f, 6>(
    vec2f(1., 0.),
    vec2f(0., 1.),
    vec2f(0., 0.), // Provoking vertex
    
    vec2f(0., 1.),
    vec2f(1., 0.),
    vec2f(1., 1.), // Provoking vertex
);
#else
const QUAD_VERTICES: array<vec2<f32>, 6> = array<vec2<f32>, 6>(
    vec2<f32>(0., 0.), // Provoking vertex
    vec2<f32>(1., 0.),
    vec2<f32>(0., 1.),
    
    vec2<f32>(1., 1.), // Provoking vertex
    vec2<f32>(0., 1.),
    vec2<f32>(1., 0.),
);
#endif
```